.github/workflows/capture-gallery-main.yml

```
@@ -0,0 +1,106 @@
+          mkdir -p docs/images
+          python scripts/capture_gallery_screenshots.py --examples examples/mvp_shiny.py --out docs/images
+
+      - name: Upload screenshots and thumbnails
+        uses: actions/upload-artifact@v4
+        with:
+          name: gallery-images
+          path: |
+            docs/images/**
+
+      - name: "Optional: Commit images to main (requires PAT)"
+        if: '${{ secrets.GALLERY_PUSH_TOKEN }}'
```

actionlint (1.7.9)

context "secrets" is not allowed here. available contexts are "env", "github", "inputs", "job", "matrix", "needs", "runner", "steps", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)